packages/flutter/lib/src/services/text_input.dart

```diff
@@ -347,6 +347,17 @@ enum TextCapitalization {
   none,
 }
 
+/// Indicates what caused the text input prompt rectangle to appear.
+enum TextInputPromptRectAppearCause {
+  /// The prompt rectangle appeared to indicate the range of text that will be
+  /// affected by autocorrection.
+  autocorrection,
```

Comment:
might be worth adding some examples to these docs (not like code examples, i mean like user interaction examples), to clarify what they're about